geode-gfsh/src/test/java/org/apache/geode/management/internal/cli/commands/StopGatewaySenderOnMemberWithBeanImplTest.java

```diff
@@ -0,0 +1,113 @@
+import org.apache.geode.management.internal.SystemManagementService;
+import org.apache.geode.test.junit.runners.GeodeParamsRunner;
+
+@RunWith(GeodeParamsRunner.class)
+public class StopGatewaySenderOnMemberWithBeanImplTest {
+  String senderId = "sender1";
+  String memberId = "member1";
+  String remoteMemberId = "remoteMember1";
+  Cache cache;
+  SystemManagementService managementService;
+  DistributedMember distributedMember;
+  DistributedMember remoteDistributedMember;
+  DistributedSystem distributedSystem;
+  GatewaySenderMXBean bean;
+
+  @Before
+  public void setUp() {
+    cache = mock(Cache.class);
+    bean = mock(GatewaySenderMXBean.class);
+    managementService = mock(SystemManagementService.class);
+    distributedMember = mock(DistributedMember.class);
+    remoteDistributedMember = mock(DistributedMember.class);
+    when(distributedMember.getId()).thenReturn(memberId);
+    when(remoteDistributedMember.getId()).thenReturn(remoteMemberId);
+    distributedSystem = mock(DistributedSystem.class);
+    doReturn(distributedSystem).when(cache).getDistributedSystem();
+  }
+
+  @Test
+  @Parameters({"true", "false"})
+  public void executeStopGatewaySenderOnMemberNotRunningReturnsNotRunningError(
+      boolean isLocalMember) {
+    StopGatewaySenderOnMember stopperWithBean = new StopGatewaySenderOnMemberWithBeanImpl();
+    setUpMocks(isLocalMember, false);
+    doReturn(false).when(bean).isRunning();
+    ArrayList<String> result = stopperWithBean.executeStopGatewaySenderOnMember(senderId, cache,
+        managementService, distributedMember);
+    assertThat(result).containsExactly(memberId, "Error",
+        "GatewaySender sender1 is not running on member " + memberId + ".");
+  }
+
+  @Test
+  @Parameters({"true", "false"})
+  public void executeStopGatewaySenderOnMemberNotAvailableReturnsNotAvailableError(
+      boolean isLocalMember) {
+    StopGatewaySenderOnMember stopperWithBean = new StopGatewaySenderOnMemberWithBeanImpl();
+    setUpMocks(isLocalMember, true);
+    ArrayList<String> result = stopperWithBean.executeStopGatewaySenderOnMember(senderId, cache,
+        managementService, distributedMember);
+    assertThat(result).containsExactly(memberId, "Error",
+        "GatewaySender sender1 is not available on member " + memberId);
+  }
+
+  @Test
+  @Parameters({"true", "false"})
+  public void executeStopGatewaySenderOnMemberRunningReturnsOk(boolean isLocalMember) {
+    StopGatewaySenderOnMember stopperWithBean = new StopGatewaySenderOnMemberWithBeanImpl();
+    setUpMocks(isLocalMember, false);
+    doReturn(true).when(bean).isRunning();
+    ArrayList<String> result = stopperWithBean.executeStopGatewaySenderOnMember(senderId, cache,
+        managementService, distributedMember);
+    assertThat(result).containsExactly(memberId, "OK",
+        "GatewaySender sender1 is stopped on member " + memberId);
+  }
+
+  private void setUpMocks(boolean isLocalMember, boolean beanMustBeNull) {
```

Comment:
How about more descriptive names like:
```
private GatewaySenderMXBean gatewaySenderMXBean(boolean isLocalMember, boolean mustBeNull) {
  GatewaySenderMXBean gatewaySenderMXBean = mustBeNull 
      ? null 
      : mock(GatewaySenderMXBean.class);
  // ...
  return gatewaySenderMXBean;
}
```
You can then delete the field `private GatewaySenderMXBean bean;` and change the tests do something like:
```
@Test
  @Parameters({"true", "false"})
  public void executeStopGatewaySenderOnMemberRunningReturnsOk(boolean isLocalMember) {
    GatewaySenderMXBean gatewaySenderMXBean = gatewaySenderMXBean(isLocalMember, false);
    when(gatewaySenderMXBean.isRunning()).thenReturn(true);

    StopGatewaySenderOnMember stopperWithBean = new StopGatewaySenderOnMemberWithBeanImpl();

    List<String> result = stopperWithBean.executeStopGatewaySenderOnMember(senderId, cache, managementService, distributedMember);

    assertThat(result)
        .containsExactly(memberId, "OK", "GatewaySender sender1 is stopped on member " + memberId);
  }
```

Reply:
Great suggestion!